### PR TITLE
fix paddle.vision.datasets.* cn docs

### LINKED ISSUE #4781: paddle.vision.datasets.Cifar10文档评估
### 文档中英文链接

中文链接：https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/vision/datasets/Cifar10_cn.html#cifar10
英文链接：https://www.paddlepaddle.org.cn/documentation/docs/en/develop/api/paddle/vision/datasets/Cifar10_en.html#cifar10

### 红线问题

1）中文的源代码中没有参数。
2）英文文档缺少“使用本API的教程文档”的英文内容。

### 结构问题

中英文文档的小标题格式不对应。

### 内容问题

_No response_

### 格式问题

示例代码中使用“import numpy as np”

### 主观评价

通过示例和说明能看懂该API接口的使用场景。 
本地环境有点问题，配置好后，提交PR。

### LINKED ISSUE #4782: paddle.vision.datasets.Cifar100文档评估
### 文档中英文链接

中文链接：https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/vision/datasets/Cifar100_cn.html#cifar100
英文链接：https://www.paddlepaddle.org.cn/documentation/docs/en/develop/api/paddle/vision/datasets/Cifar100_en.html#cifar100

### 红线问题

1）中文的源代码中没有参数。
2）英文文档缺少对参数data_file (str) 中“data_file 参数可以设置为 None”这句话的翻译

### 结构问题

中英文文档的小标题格式不对应。

### 内容问题

_No response_

### 格式问题

_No response_

### 主观评价

通过示例和说明能看懂该API接口的使用场景。 
本地环境有点问题，配置好后，提交PR。

### LINKED ISSUE #4784: paddle.vision.datasets.ImageFolder文档评估
### 文档中英文链接

中文链接：https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/vision/datasets/ImageFolder_cn.html#imagefolder
英文链接：https://www.paddlepaddle.org.cn/documentation/docs/en/develop/api/paddle/vision/datasets/ImageFolder_en.html#imagefolder

### 红线问题

1）英文文档“root/1.ext root/2.ext root/sub_dir/3.ext”多余语句。
2）英文文档参数extensions缺少对“默认使用 ('.jpg', '.jpeg', '.png', '.ppm', '.bmp', '.pgm', '.tif', '.tiff', '.webp') ”内容的介绍。
3）中英文示例代码不一样。

### 结构问题

_No response_

### 内容问题

_No response_

### 格式问题

示例代码中使用“import numpy as np”

### 主观评价

通过示例和说明能看懂该API接口的使用场景。 
本地环境有点问题，配置好后，提交PR。

### LINKED ISSUE #4790: paddle.vision.datasets.MNIST文档评估
### 文档中英文链接

中文链接：https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/vision/datasets/MNIST_cn.html#mnist
英文链接：https://www.paddlepaddle.org.cn/documentation/docs/en/develop/api/paddle/vision/datasets/MNIST_en.html#mnist

### 红线问题

1）中文文档中源代码缺少“(image_path=None, label_path=None, mode='train', transform=None, download=True, backend=None)”

### 结构问题

_No response_

### 内容问题

中英文文档的小标题格式不对应。

### 格式问题

_No response_

### 主观评价

功能性接口，用法简单，通过示例和说明能看懂该API接口的使用场景。
本地环境有点问题，配置好后，提交PR。

### LINKED ISSUE #4791: paddle.vision.datasets.VOC2012文档评估
### 文档中英文链接

中文链接：https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/vision/datasets/VOC2012_cn.html#voc2012
英文链接：https://www.paddlepaddle.org.cn/documentation/docs/en/develop/api/paddle/vision/datasets/VOC2012_en.html#voc2012

### 红线问题

1）中文文档中源代码缺少“(data_file=None, mode='train', transform=None, download=True, backend=None)”
2）英文文档多了“To speed up the download, we put the data on https://dataset.bj.bcebos.com/voc/VOCtrainval_11-May-2012.tar. Original data can get from http://host.robots.ox.ac.uk/pascal/VOC/voc2012/VOCtrainval_11-May-2012.tar.”

### 结构问题

_No response_

### 内容问题

中英文文档的小标题格式不对应。

### 格式问题

_No response_

### 主观评价

功能性接口，用法简单，通过示例和说明能看懂该API接口的使用场景。
本地环境有点问题，配置好后，提交PR。

### LINKED ISSUE #4792: paddle.vision.datasets.Flowers文档评估
### 文档中英文链接

中文链接：https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/vision/datasets/Flowers_cn.html#flowers
英文链接：https://www.paddlepaddle.org.cn/documentation/docs/en/develop/api/paddle/vision/datasets/Flowers_en.html#flowers

### 红线问题

1）中文文档中源代码缺少“(data_file=None, label_file=None, setid_file=None, mode='train', transform=None, download=True, backend=None)”

### 结构问题

_No response_

### 内容问题

中英文文档的小标题格式不对应。

### 格式问题

_No response_

### 主观评价

功能性接口，用法简单，通过示例和说明能看懂该API接口的使用场景。
本地环境有点问题，配置好后，提交PR。

### LINKED ISSUE #43194: paddle.vision.datasets.MNIST API文档和源代码以及官方使用示例不一致
### 请提出你的问题 Please ask your question

MNIST文档内容如下：
image
可是我在一些官方实例中看到多了个transform参数，
```python
from paddle.vision.transforms import Normalize

# 定义图像归一化处理方法，这里的CHW指图像格式需为 [C通道数，H图像高度，W图像宽度]
transform = Normalize(mean=[127.5], std=[127.5], data_format='CHW')
# 下载数据集并初始化 DataSet
train_dataset = paddle.vision.datasets.MNIST(mode='train', transform=transform)
test_dataset = paddle.vision.datasets.MNIST(mode='test', transform=transform)
print('train images: ',len(train_dataset),', test images: ',len(test_dataset))
```
我知道这个是用来处理图像，但是API文档里没有这个。

我看了下class paddle.vision.datasets.MNIST[[源代码]](https://github.com/PaddlePaddle/Paddle/blob/release/2.3/python/paddle/vision/datasets/mnist.py#L30)
这里面是有这个参数的，API文档里应该是忘了加上相应的描述？

```python
 def __init__(self,
                 image_path=None,
                 label_path=None,
                 mode='train',
                 transform=None,
                 download=True,
                 backend=None):
        assert mode.lower() in ['train', 'test'], \
                "mode should be 'train' or 'test', but got {}".format(mode)

        if backend is None:
            backend = paddle.vision.get_image_backend()
        if backend not in ['pil', 'cv2']:
            raise ValueError(
                "Expected backend are one of ['pil', 'cv2'], but got {}"
                .format(backend))
        self.backend = backend

        self.mode = mode.lower()
        self.image_path = image_path
        if self.image_path is None:
            assert download, "image_path is not set and downloading automatically is disabled"
            image_url = self.TRAIN_IMAGE_URL if mode == 'train' else self.TEST_IMAGE_URL
            image_md5 = self.TRAIN_IMAGE_MD5 if mode == 'train' else self.TEST_IMAGE_MD5
            self.image_path = _check_exists_and_download(
                image_path, image_url, image_md5, self.NAME, download)

        self.label_path = label_path
        if self.label_path is None:
            assert download, "label_path is not set and downloading automatically is disabled"
            label_url = self.TRAIN_LABEL_URL if self.mode == 'train' else self.TEST_LABEL_URL
            label_md5 = self.TRAIN_LABEL_MD5 if self.mode == 'train' else self.TEST_LABEL_MD5
            self.label_path = _check_exists_and_download(
                label_path, label_url, label_md5, self.NAME, download)

        self.transform = transform
```
我是一个刚接触paddle和python的小白，不知道理解的对不对。

### PR DESCRIPTION
**Pull request by SigureMo**
## PR 内容

paddle.vision.datasets API 文档修复

## 英文文档同步更新 PR

https://github.com/PaddlePaddle/Paddle/pull/43649

PADDLEPADDLE_PR=43649

## 相关 issue

- https://github.com/PaddlePaddle/Paddle/issues/43194
- #4792
- #4791
- #4790
- #4784
- #4782
- #4781

总结存在的问题：

- [x] ~~代码示例缺少输出~~代码示例不统一，且无法表现各个 API 的特性及共性，全部重写
- [x] ~~部分使用 numpy 的示例~~同上
- [x] 优化返回值部分
- [x] optional 和 可选
- [x] 句首字母大写
- [x] 中文介绍没有和函数定义间隔一行
- [x] 中英文使用空格隔开的规范问题
- [x] 部分中文文档缺少参数表
- [x] 「MNIST、VOC2012」缺少参数 `transform`
- [x] ~~「VOC2012」的提升速度提示应作为 Note，然后中文同步增加该内容~~因为所有的数据集都不是从原始来源下载，因此这个说明显得有点多余，故删除
- [x] 「ImageFolder」中文文档描述部分缺少内容
- [x] 「ImageFolder」参数 extensions 缺少部分内容

可 close 的 issue：

closes https://github.com/PaddlePaddle/Paddle/issues/43194
fixes #4792, fixes #4791, fixes #4790, fixes #4784, fixes #4782, fixes #4781

## 其它

另外因为发现之前修复的 paddle.Model API 时候 COPY-FROM 的规则搞错了，我原以为同一个类所有示例共享一个作用域需要用 label 分开，原来类下方法有自己的作用域，因此修复了下～

~~另外这个 PR 没做 COPY-FROM，所以中文文档是旧的示例代码，等 #4940 自动更新了~~ 合入的 #4949 莫名其妙没更新 paddle.vision.datasets 下面的……还要我手动更新……

### CODE REVIEW

**Comment by Ligoml** [comment]
COPY-FROM 已通过https://github.com/PaddlePaddle/docs/pull/4949 完成统一更新~

File: docs/api/paddle/vision/datasets/DatasetFolder_cn.rst [hunk 1]
```diff
@@ -5,26 +5,40 @@ DatasetFolder
...
 
- 一种通用的数据加载方式，输入以如下的格式存放：
+
+一种通用的数据加载方式，数据需要以如下的格式存放：
 
 .. code-block:: text
- root/class_a/1.ext
- root/class_a/2.ext
- root/class_a/3.ext
+    root/class_a/1.ext
```
Comment (Ligoml) [inline docs/api/paddle/vision/datasets/DatasetFolder_cn.rst:13]: .. code-block:: text 与内容之间需要空一行，否则会无法显示

File: docs/api/paddle/vision/datasets/DatasetFolder_cn.rst [hunk 2]
```diff
@@ -5,26 +5,40 @@ DatasetFolder
...
+  - **extensions** (list[str]|tuple[str]，可选) - 允许的数据后缀列表，``extensions`` 和 ``is_valid_file`` 不可以同时设置。如果该值没有设定，默认为 ``('.jpg', '.jpeg', '.png', '.ppm', '.bmp', '.pgm', '.tif', '.tiff', '.webp')``。默认值为 None。
+  - **transform** (Callable，可选) - 图片数据的预处理，若为 ``None`` 即为不做预处理。默认值为 ``None``。
+  - **is_valid_file** (Callable，可选) - 根据每条数据的路径来判断是否合法的一个函数。``extensions`` 和 ``is_valid_file`` 不可以同时设置。默认值为 None。
+
+返回
+:::::::::
+
+:ref:`cn_api_io_cn_Dataset`，DatasetFolder 实例。
+
+属性
```
Comment (Ligoml) [inline docs/api/paddle/vision/datasets/DatasetFolder_cn.rst:36]: 属性部分请参考[规范](https://github.com/PaddlePaddle/docs/wiki/%E9%A3%9E%E6%A1%A8API%E6%96%87%E6%A1%A3%E4%B9%A6%E5%86%99%E8%A7%84%E8%8C%83#api-%E5%B1%9E%E6%80%A7)

**Comment by SigureMo** [comment]
> COPY-FROM 已通过#4949 完成统一更新~

结果 #4949 没有更新这一部分（paddle.vision.datasets.*） :joy:，明明 #4940 里更新了的说，不过我手动更新完了～

File: docs/api/paddle/vision/datasets/DatasetFolder_cn.rst (hunk 2 above)
Comment (SigureMo) [reply docs/api/paddle/vision/datasets/DatasetFolder_cn.rst:36]: 刚刚看了一下，但关于属性我有很大的疑问，这个属性如果中文文档这么写的话，英文文档要往哪里写呢？因为属性是不同于方法没有一个地方写 docstring 的呀（应该是这样子的吧）？

我看了下示例的 [paddle.static.Program](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/static/Program_cn.html)（paddle.Program 没有找到，貌似是 API 改名了？），我发现文档里暴露的属性都是 `@property` 装饰（或者说 getter）后的属性，这些属性确实是可以写 docstring，但相应的，非 `@property` 装饰的属性要如何写 docstring 呢？而且这个 API 文档里貌似有点混乱，属性里混入了大量的方法

image

唔，我又翻看了一些其他的类，貌似并没有找到相关可以参考的内容（非 `@property` 装饰的暴露的属性且中英文文档都完善的）～所以说属性的话是不管英文文档嘛？

**Comment by Ligoml** [comment]
> > COPY-FROM 已通过#4949 完成统一更新~
> 
> 结果 #4949 没有更新这一部分（paddle.vision.datasets.*） 😂，明明 #4940 里更新了的说，不过我手动更新完了～

好叭，#4940 因为出现了一些bad case所以close了重提的 #4949，改了就好哈哈

File: docs/api/paddle/vision/datasets/DatasetFolder_cn.rst (hunk 2 above)
Comment (Ligoml) [reply docs/api/paddle/vision/datasets/DatasetFolder_cn.rst:36]: @TCChenlong 呼唤龙哥，这个标准怎么说？

**Review by Ligoml — APPROVED** [review]
LGTM
